components/common-go/pprof/pprof.go

```diff
@@ -33,55 +29,13 @@ func Serve(addr string) {
 // Handler produces the pprof endpoint handler
 func Handler() *http.ServeMux {
 	mux := http.NewServeMux()
-	mux.HandleFunc(Path, index)
+	mux.HandleFunc(Path, pprof.Index)
 	mux.HandleFunc(Path+"cmdline", pprof.Cmdline)
 	mux.HandleFunc(Path+"profile", pprof.Profile)
 	mux.HandleFunc(Path+"symbol", pprof.Symbol)
 	mux.HandleFunc(Path+"trace", pprof.Trace)
-	return mux
-}
-
-func index(w http.ResponseWriter, r *http.Request) {
-	if strings.HasPrefix(r.URL.Path, Path) {
-		// according to Ian Lance Taylor it's ok to turn on mutex and block profiling
-		// when asking for the actual profile [1]. This handler implements this idea, as
-		// discussed in [2]
-		//
-		// [1] https://groups.google.com/forum/#!topic/golang-nuts/qiHa97XzeCw
-		// [2] https://github.com/golang/go/issues/23401
-
-		var (
-			name          = strings.TrimPrefix(r.URL.Path, Path)
-			seconds, serr = strconv.ParseInt(r.URL.Query().Get("seconds"), 10, 64)
-		)
-		if name == "mutex" {
-			frac, ferr := strconv.ParseInt(r.URL.Query().Get("frac"), 10, 64)
-			if serr == nil && ferr == nil && seconds > 0 && frac > 0 {
-				//nolint:gosec
-				id := rand.Uint32()
-				log.WithField("id", id).WithField("frac", frac).WithField("seconds", seconds).Debug("enabled mutex profiling")
 
-				runtime.SetMutexProfileFraction(int(frac))
-				defer func() {
-					runtime.SetMutexProfileFraction(0)
-					log.WithField("id", id).WithField("frac", frac).WithField("seconds", seconds).Debug("disabled mutex profiling")
-				}()
-			}
```

Comment:
I'm afraid that by removing this we loose the ability to set `frac` at runtime.
https://github.com/golang/go/commit/0ee4b13830572ad66cdcc045d1e4ae1e15a87ebb only adds `seconds`

Reply:
reverted